### PR TITLE
[JENKINS-52869] Show error messages on user creation

### PR DESCRIPTION
**Pull request by daniel-beck**
See [JENKINS-52869](https://issues.jenkins-ci.org/browse/JENKINS-52869).

> screen shot

Someone else can make this as fancy as the redesigned regular signup, for now we need the error messages back.

### Proposed changelog entries

* Bug: User account creation by administrators did not show error messages when it failed. (Regression in 2.129).

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

**Review by jvz — APPROVED** [review]

**Review by fcojfernandez — APPROVED** [review]

**Review by Wadeck — APPROVED** [review]
:bee:

**Comment by daniel-beck** [comment]
For some reason the old `errorMessage` field still exists, so we probably want to retain compatibility with it.

**Review by varyvol — APPROVED** [review]

**Comment by batmat** [comment]
@daniel-beck shoud you assign yourself to https://issues.jenkins-ci.org/browse/JENKINS-52869 ? Or is this PR only part of it or so?

File: core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/_entryForm.jelly [hunk 1]
```diff
@@ -27,11 +27,11 @@ THE SOFTWARE.
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
   <h1>${title}</h1>
   <div style="margin: 2em;">
-    <j:if test="${data.errorMessage!=null}">
+    <j:forEach var="error" items="${data.errors}">
```
Comment (batmat) [inline core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/_entryForm.jelly]: OK, so IIUC in https://github.com/jenkinsci/jenkins/commit/90670e05c17b34270cda9436eaf877cc72a341ec#diff-de6af001ecaa93f85c803b52f8831a99R550 `errorMessage` got enriched with an `errors` collection, but this got overlooked.

BTW, I see there we left the `errorMessage` variable in the `SignupInfo` class. Is this for backward compatibility, or is this another leftover that should also be cleaned up?

**Review by batmat — APPROVED** [review]
LGTM, one comment/question.

**Comment by daniel-beck** [comment]
Considering this finished.

**Comment by batmat** [comment]
FTR current test failures (known issue and unrelated)
screenshot_20181209-094057__01

**Review by batmat — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Missed the PR. I think we can land it into this weekly

**Pull request merged**